File: src/com/woezelmann/showas/ShowAsFromConsole.java
package com.woezelmann.showas;

import com.intellij.lang.Language;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.editor.Editor;

public class ShowAsFromConsole extends ShowAs {

    public ShowAsFromConsole(Language language) {
        super(language);
    }

    @Override
    public String fetchValue(AnActionEvent event) {
        Editor data = event.getData(PlatformDataKeys.EDITOR);
        if (data != null) {
            return data.getSelectionModel().getSelectedText();
        }

        return null;
    }
}

File: src/main/kotlin/com/woezelmann/showas/console/ShowAsFromConsole.kt
package com.woezelmann.showas.console

import com.intellij.lang.Language
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.PlatformDataKeys
import com.intellij.openapi.editor.Editor
import com.woezelmann.showas.ShowAs


open class ShowAsFromConsole(language: Language) : ShowAs(language) {
    override fun fetchValue(event: AnActionEvent?): String? {
        val data: Editor? = event!!.getData(PlatformDataKeys.EDITOR)
        return data?.getSelectionModel()?.getSelectedText()
    }
}